### ISSUE TITLE
bug: Ionic 6 - Ion-Datetime UI not adding local timezone hrs to passed date

### ISSUE THREAD

**Issue by bkarv** [ionic-team/ionic-framework#24458]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

Upgraded to Ionic 6 and have noticed an issue when passing a datetime to Ionic DateTime.  Below is the following UTC datetime: 
`2021-12-16T18:54:00+00:00`

I am currently in Australian time which is +10hrs from the UTC. This means when I format the date using Angular's inbuilt pipefunction to a medium date (ie only showing the date no time), it adds my timezone 10hrs, which pushes the date into the next day and displays it as:
`Dec 17, 2021`

I store this date in a variable and then pass it to the ion-datetime via ngModel. The issue is I expect when I open the picker the date to automatically add the 10+ hrs and display the solid circle on the 17th of Dec. However instead it uses the UTC timezone which shows the datepicker on the 16th of Dec instead.
<img width="333" alt="Screen Shot 2021-12-21 at 10 26 13 pm" src="https://user-images.githubusercontent.com/10624315/146922566-c7238b05-b001-4340-86b6-8afe87bde63e.png">

### Expected Behavior

As in Ionic 5, the picker for date
`2021-12-16T18:54:00+00:00`

and where my timezone is 10+hrs of the UTC time, it should as 17th Dec 2021. Ionic 5 does this correctly.
<img width="307" alt="Screen Shot 2021-12-21 at 10 19 52 pm" src="https://user-images.githubusercontent.com/10624315/146922688-98c1f69e-42b4-42c6-9ddc-71c00e6dadc3.png">
<img width="503" alt="Screen Shot 2021-12-21 at 10 19 44 pm" src="https://user-images.githubusercontent.com/10624315/146922705-bc2a78ec-8e9f-4ab4-8a61-9f6218db6955.png">

### Steps to Reproduce

1. Set a datetime to a variable where in UTC it would land on a different day to your local timezone. Eg for people in California with UTC -8hrs, it would be:
`testDate = "2021-12-16T01:54:00+00:00"`

2. Pass variable to picker and check if the proper day is selected. For above example for Californians it should be the 15th of Dec. If having same bug however, it will be displayed as 16th of Dec. 

```
<ion-item class="ion-text-wrap" button="true" id="ionic-date-contact-start">
                        <ion-icon name="calendar" slot="start"></ion-icon>
                          <ion-label style="color:#424242">Start Date</ion-label>
                          <ion-text>{{testDate|date:'mediumDate'}}</ion-text>
                        <ion-popover trigger="ionic-date-contact-start" show-backdrop="false">
                          <ng-template>
                            <ion-datetime
                              #popoverDatetime
                              presentation="date"
                              [(ngModel)]="testDate"
                              [showClearButton]="true"
                            ></ion-datetime>
                          </ng-template>
                        </ion-popover>
                      </ion-item>
```

### Ionic Info

Ionic:

   Ionic CLI                     : 6.15.0 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 6.0.1
   @angular-devkit/build-angular : 13.0.2
   @angular-devkit/schematics    : 13.0.2
   @angular/cli                  : 13.0.2
   @ionic/angular-toolkit        : 5.0.3

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : none
   Cordova Plugins   : no whitelisted plugins (0 plugins total)

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v16.13.1 (/usr/local/bin/node)
   npm    : 8.1.1
   OS     : macOS Big Sur
   Xcode  : Xcode 13.2.1 Build version 13C100

**Comment by liamdebeasi** [comment]
Thanks for the issue. This behavior is by design. The datetime in Ionic 6 makes no assumptions about the timezone in your app. Previously in Ionic 5, there were many issues with timezones being converted automatically/incorrectly that impacted the user experience and provided for unexpected behaviors.

I did a quick check, and the native datetime implementation in browsers has the same behavior: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#setting_timezones

I added some examples to the docs on how to format dates according to the local timezone that you might find helpful: https://github.com/ionic-team/ionic-framework/pull/24425

I am going to close this as this is not a bug In Ionic. Thanks!
